<a href="https://colab.research.google.com/github/hiba0903/DSA_CS_Ast/blob/main/DSA_DataAq_CS.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

○ Load the dataset into a pandas DataFrame.


In [ ]:
import pandas as pd  # Import the pandas library

df = pd.read_csv("/content/Food Nutrition Dataset.csv")
print(df)

     Unnamed: 0.1  Unnamed: 0                      food  Caloric Value  \
0               0           0    margarine with yoghurt             88   
1               1           1     sunflower seed butter             99   
2               2           2              hazelnut oil            120   
3               3           3         menhaden fish oil           1966   
4               4           4        cod liver fish oil            123   
..            ...         ...                       ...            ...   
717           717         717                  jews ear             25   
718           718         718           enoki mushrooms              1   
719           719         719           morel mushrooms              4   
720           720         720  portabella mushrooms raw             19   
721           721         721           oyster mushroom              5   

         Fat  Saturated Fats  Monounsaturated Fats  Polyunsaturated Fats  \
0      9.800           1.900       

Display the first few rows of the DataFrame to understand its structure.

In [ ]:
print(df.head())

   Unnamed: 0.1  Unnamed: 0                    food  Caloric Value    Fat  \
0             0           0  margarine with yoghurt             88    9.8   
1             1           1   sunflower seed butter             99    8.8   
2             2           2            hazelnut oil            120   13.6   
3             3           3       menhaden fish oil           1966  218.0   
4             4           4      cod liver fish oil            123   13.6   

   Saturated Fats  Monounsaturated Fats  Polyunsaturated Fats  Carbohydrates  \
0             1.9                   5.6                   2.0          0.073   
1             0.7                   6.2                   1.6          3.700   
2             1.0                  10.6                   1.4          0.000   
3            66.3                  58.2                  74.5          0.000   
4             3.1                   6.4                   3.1          0.000   

   Sugars  ...  Calcium  Copper   Iron  Magnesium  Manga

Print the summary statistics of the dataset.


In [ ]:
print(df.info())

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 722 entries, 0 to 721
Data columns (total 37 columns):
 #   Column                Non-Null Count  Dtype  
---  ------                --------------  -----  
 0   Unnamed: 0.1          722 non-null    int64  
 1   Unnamed: 0            722 non-null    int64  
 2   food                  722 non-null    object 
 3   Caloric Value         722 non-null    int64  
 4   Fat                   722 non-null    float64
 5   Saturated Fats        722 non-null    float64
 6   Monounsaturated Fats  722 non-null    float64
 7   Polyunsaturated Fats  722 non-null    float64
 8   Carbohydrates         722 non-null    float64
 9   Sugars                722 non-null    float64
 10  Protein               722 non-null    float64
 11  Dietary Fiber         722 non-null    float64
 12  Cholesterol           722 non-null    float64
 13  Sodium                722 non-null    float64
 14  Water                 722 non-null    float64
 15  Vitamin A             7

Q1: What is the size of the dataset? (Number of rows and columns)

In [ ]:
#retrieving rows and columns using shape()

row,col = df.shape
print(row,"x",col)

722 x 37


2: What are the data types of each column?

In [ ]:
#dtypes returns the datatypes

print(df.dtypes)

Unnamed: 0.1              int64
Unnamed: 0                int64
food                     object
Caloric Value             int64
Fat                     float64
Saturated Fats          float64
Monounsaturated Fats    float64
Polyunsaturated Fats    float64
Carbohydrates           float64
Sugars                  float64
Protein                 float64
Dietary Fiber           float64
Cholesterol             float64
Sodium                  float64
Water                   float64
Vitamin A               float64
Vitamin B1              float64
Vitamin B11             float64
Vitamin B12             float64
Vitamin B2              float64
Vitamin B3              float64
Vitamin B5              float64
Vitamin B6              float64
Vitamin C               float64
Vitamin D               float64
Vitamin E               float64
Vitamin K               float64
Calcium                 float64
Copper                  float64
Iron                    float64
Magnesium               float64
Manganes

Are there any missing values in the dataset? If so, which columns have
missing values and how many?

In [ ]:
# Count the missing in each cols
missing = df.isnull().sum()

#cols that have no missing values
null_cols = missing[missing > 0]

# Print the columns with missing values and their counts
print(null_cols)

#the series returns the cols with null values aling with th count

     Unnamed: 0.1  Unnamed: 0   food  Caloric Value    Fat  Saturated Fats  \
0           False       False  False          False  False           False   
1           False       False  False          False  False           False   
2           False       False  False          False  False           False   
3           False       False  False          False  False           False   
4           False       False  False          False  False           False   
..            ...         ...    ...            ...    ...             ...   
717         False       False  False          False  False           False   
718         False       False  False          False  False           False   
719         False       False  False          False  False           False   
720         False       False  False          False  False           False   
721         False       False  False          False  False           False   

     Monounsaturated Fats  Polyunsaturated Fats  Carbohydrates 

What are the unique values in a categorical column of your choice? How
many unique values are there?

In [ ]:
# categorical column selected
cat_col = 'food'

# Get unique values
uni_val = df[cat_col].unique()

# Count of unique values
num_uni_val = len(uni_val)

# Print the unique values and their count
print("Unique values in ",cat_col)
print("\n",uni_val)
print("\nNumber of unique values: ",num_uni_val)


Unique values in  food

 ['margarine with yoghurt' 'sunflower seed butter' 'hazelnut oil'
 'menhaden fish oil' 'cod liver fish oil' 'sardine oil' 'cupu assu oil'
 'lard' 'sesame butter' 'margarine spread' 'lard vegetable oil'
 'shortening' 'vegetable oil spread' 'cottonseed oil'
 'vegetable oil spread fat free' 'soy margarine' 'margarine with salt'
 'margarine' 'salmon fish oil' 'nutmeg butter oil' 'almond butter'
 'sour cream shoprite' 'ucuhuba butter oil' 'chicken fat'
 'herring fish oil' 'bacon grease' 'turkey fat' 'beef tallow' 'goose fat'
 'duck fat' 'mutton tallow' 'grapeseed oil' 'sheanut oil' 'almond oil'
 'sunflower oil' 'cashew butter' 'canola oil' 'babassu oil' 'teaseed oil'
 'avocado oil' 'vegetable palm kernel oil' 'coconut oil' 'safflower oil'
 'vegetable oil' 'apricot kernel oil' 'peanut oil' 'poppyseed oil'
 'flaxseed oil' 'tomatoseed oil' 'wheat germ oil' 'sesame oil'
 'corn granola oil' 'cocoa butter oil' 'olive oil' 'palm kernel oil'
 'mustard oil' 'rice bran oil' 'v

There are 722 unique values in the column food

 **For a numerical column of your choice, calculate the mean, median, and
standard deviation.**


In [ ]:
#selecting numerical column
num_col = 'Vitamin K'

# Calculating mean
print("Mean:", df[num_col].mean())

#Median
print("Median:", df[num_col].median())

#Standard Deviation
print("Standard Deviation:", df[num_col].std())

Mean: 0.18084487534626043
Median: 0.031
Standard Deviation: 0.6322358961101077


**Identify any potential outliers in a numerical column of your choice. Explain
your approach.**

In [ ]:
#selecting numerical column
num_col = 'Zinc'

# Calculate Q1 (25th percentile) and Q3 (75th percentile)
Q1 = df[num_col].quantile(0.25)
Q3 = df[num_col].quantile(0.75)

# Calculate the IQR
IQR = Q3 - Q1
print("IQR = ",IQR)
# Determine the lower and upper bounds for outliers
print("\nLower Bound",Q1 - 1.5 * IQR)
print("\nUpper Bound", Q3 + 1.5 * IQR)

# Identify outliers
outliers = df[(df[num_col] < lower_bound) | (df[num_col] > upper_bound)]

# Print the outliers
print("\nOutliers in ",num_col)
print(outliers)


IQR =  0.365

Lower Bound -0.5125

Upper Bound 0.9475

Outliers in  Zinc
     Unnamed: 0.1  Unnamed: 0                      food  Caloric Value   Fat  \
8               8           8             sesame butter             94   8.1   
61             61          61            corn pasta dry            375   2.2   
69             69          69  whole wheat macaroni dry            365   1.5   
71             71          71              macaroni raw            390   1.6   
75             75          75              pasta cooked            301   2.4   
..            ...         ...                       ...            ...   ...   
666           666         666          asparagus canned             62   0.7   
671           671         671            spinach canned             44   0.9   
672           672         672       sweet potato canned            568   1.3   
686           686         686       tomato paste canned            215   1.2   
693           693         693      taco salad t

Any values lower than lower bound and greater than upperbound are outliers

**Find the correlation between numerical columns and discuss any interesting
relationships.**

In [ ]:
# Selecting numerical column
numerical_df = df.select_dtypes(include=['number'])

# Calculating correlation matrix
correlation_matrix = numerical_df.corr()

# correlation matrix
print("Correlation Matrix:")
print(correlation_matrix)


Correlation Matrix:
                      Unnamed: 0.1  Unnamed: 0  Caloric Value       Fat  \
Unnamed: 0.1              1.000000    1.000000      -0.243587 -0.230824   
Unnamed: 0                1.000000    1.000000      -0.243587 -0.230824   
Caloric Value            -0.243587   -0.243587       1.000000  0.880499   
Fat                      -0.230824   -0.230824       0.880499  1.000000   
Saturated Fats           -0.172714   -0.172714       0.692775  0.843804   
Monounsaturated Fats     -0.259062   -0.259062       0.738356  0.812784   
Polyunsaturated Fats     -0.169602   -0.169602       0.675207  0.697706   
Carbohydrates            -0.116083   -0.116083       0.573389  0.132991   
Sugars                    0.089851    0.089851       0.157132 -0.024324   
Protein                   0.029889    0.029889       0.423404  0.152858   
Dietary Fiber             0.225389    0.225389       0.315062  0.042817   
Cholesterol              -0.081507   -0.081507       0.338563  0.406905   
Sodiu

**+1**: Perfect positive correlation (as one variable increases, the other also increases).

**-1**: Perfect negative correlation (as one variable increases, the other decreases).

**0**: No correlation (no linear relationship between the variables).

  -0.030915 is the correlation of Sugar and Saturated Fats.This shows that as the amount of sugar increases ,the saturated fat per intake decreases

   -0.002799 is between manganese and calorific value. Since the value is near to 0 there is no significant relation
   
    

**8: Group the data by a categorical column and compute the aggregate statistics
(e.g., mean, sum) for a numerical column.**

In [ ]:
# Group by the categorical column 'Category' and compute the mean and sum for 'Value'
aggregate_stats = df.groupby('food')['Nutrition Density'].agg(['mean', 'sum'])

print("Aggregate Statistics:")
print(aggregate_stats)


Aggregate Statistics:
                        mean      sum
food                                 
acorn squash cooked  155.700  155.700
acorn squash raw     247.800  247.800
adobo fresco         503.400  503.400
agar                   0.587    0.587
almond butter         73.000   73.000
...                      ...      ...
yellow tomato         10.800   10.800
yuca cassava chips     2.621    2.621
zucchini cooked       65.600   65.600
zucchini raw          15.151   15.151
zwieback               7.083    7.083

[722 rows x 2 columns]


## Q9:: Based on your analysis, provide a brief summary of any insights or patterns you discovered in the dataset.



*   722 x 37 is the size of dataset
*   There are 722 unique values in the column food
*   A list of food items that are outliers based on their Zinc content, compared to the overall data distribution. Foods like "sesame butter" and "taco salad taco bell" have been flagged as outliers.
*   -0.030915 is the correlation of Sugar and Saturated Fats.This shows that as the amount of sugar increases ,the saturated fat per intake decreases

-0.002799 is between manganese and calorific value. Since the value is near to 0 there is no significant relation



